### ISSUE TITLE
Project dependencies

### ISSUE THREAD

**Issue by clausmichele** [Open-EO/openeo-processes-dask#60]
Working on the Client Side Processing we realized that using `openeo-processes-dask` and `openeo-pg-parser-networkx` drags in many dependencies as @soxofaan noted here: 
https://github.com/Open-EO/openeo-python-client/pull/338#issuecomment-1418936710

Could we check if all of them are really used? ~~`datacube` for instance is not used directly but still listed as requirement.~~

**Comment by clausmichele** [comment]
Ok here is the list of all the current dependencies, I see that `datacube` is anyway required by `odc-algo`:
```
openeo-processes-dask==2023.1.2
  - dask [required: >=2022.11.1, installed: 2022.12.0]
    - click [required: >=7.0, installed: 8.1.3]
      - colorama [required: Any, installed: 0.4.6]
    - cloudpickle [required: >=1.1.1, installed: 2.2.0]
    - fsspec [required: >=0.6.0, installed: 2022.11.0]
    - packaging [required: >=20.0, installed: 21.3]
      - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
    - partd [required: >=0.3.10, installed: 1.3.0]
      - locket [required: Any, installed: 1.0.0]
      - toolz [required: Any, installed: 0.12.0]
    - pyyaml [required: >=5.3.1, installed: 6.0]
    - toolz [required: >=0.8.2, installed: 0.12.0]
  - dask-geopandas [required: >=0.2.0,<1, installed: 0.2.0]
    - dask [required: >=2021.06.0, installed: 2022.12.0]
      - click [required: >=7.0, installed: 8.1.3]
        - colorama [required: Any, installed: 0.4.6]
      - cloudpickle [required: >=1.1.1, installed: 2.2.0]
      - fsspec [required: >=0.6.0, installed: 2022.11.0]
      - packaging [required: >=20.0, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - partd [required: >=0.3.10, installed: 1.3.0]
        - locket [required: Any, installed: 1.0.0]
        - toolz [required: Any, installed: 0.12.0]
      - pyyaml [required: >=5.3.1, installed: 6.0]
      - toolz [required: >=0.8.2, installed: 0.12.0]
    - distributed [required: >=2021.06.0, installed: 2022.12.0]
      - click [required: >=7.0, installed: 8.1.3]
        - colorama [required: Any, installed: 0.4.6]
      - cloudpickle [required: >=1.5.0, installed: 2.2.0]
      - dask [required: ==2022.12.0, installed: 2022.12.0]
        - click [required: >=7.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
        - cloudpickle [required: >=1.1.1, installed: 2.2.0]
        - fsspec [required: >=0.6.0, installed: 2022.11.0]
        - packaging [required: >=20.0, installed: 21.3]
          - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
        - partd [required: >=0.3.10, installed: 1.3.0]
          - locket [required: Any, installed: 1.0.0]
          - toolz [required: Any, installed: 0.12.0]
        - pyyaml [required: >=5.3.1, installed: 6.0]
        - toolz [required: >=0.8.2, installed: 0.12.0]
      - jinja2 [required: Any, installed: 3.1.2]
        - MarkupSafe [required: >=2.0, installed: 2.1.1]
      - locket [required: >=1.0.0, installed: 1.0.0]
      - msgpack [required: >=0.6.0, installed: 1.0.4]
      - packaging [required: >=20.0, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - psutil [required: >=5.0, installed: 5.9.4]
      - pyyaml [required: Any, installed: 6.0]
      - sortedcontainers [required: !=2.0.1,!=2.0.0, installed: 2.4.0]
      - tblib [required: >=1.6.0, installed: 1.7.0]
      - toolz [required: >=0.10.0, installed: 0.12.0]
      - tornado [required: >=6.0.3, installed: 6.2]
      - urllib3 [required: Any, installed: 1.26.13]
      - zict [required: >=0.1.3, installed: 2.2.0]
        - heapdict [required: Any, installed: 1.0.1]
    - geopandas [required: >=0.10, installed: 0.12.1]
      - fiona [required: >=1.8, installed: 1.8.22]
        - attrs [required: >=17, installed: 22.1.0]
        - certifi [required: Any, installed: 2022.12.7]
        - click [required: >=4.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
        - click-plugins [required: >=1.0, installed: 1.1.1]
          - click [required: >=4.0, installed: 8.1.3]
            - colorama [required: Any, installed: 0.4.6]
        - cligj [required: >=0.5, installed: 0.7.2]
          - click [required: >=4.0, installed: 8.1.3]
            - colorama [required: Any, installed: 0.4.6]
        - munch [required: Any, installed: 2.5.0]
          - six [required: Any, installed: 1.16.0]
        - setuptools [required: Any, installed: 65.5.1]
        - six [required: >=1.7, installed: 1.16.0]
      - packaging [required: Any, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - pandas [required: >=1.0.0, installed: 1.5.2]
        - numpy [required: >=1.20.3, installed: 1.24.1]
        - python-dateutil [required: >=2.8.1, installed: 2.8.2]
          - six [required: >=1.5, installed: 1.16.0]
        - pytz [required: >=2020.1, installed: 2022.6]
      - pyproj [required: >=2.6.1.post1, installed: 3.4.0]
        - certifi [required: Any, installed: 2022.12.7]
      - shapely [required: >=1.7, installed: 1.8.4]
    - packaging [required: Any, installed: 21.3]
      - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
    - pygeos [required: Any, installed: 0.13]
      - numpy [required: >=1.13, installed: 1.24.1]
  - datacube [required: >=1.8.4,<2.0.0, installed: 1.8.9]
    - affine [required: Any, installed: 2.3.1]
    - cachetools [required: Any, installed: 5.2.0]
    - click [required: >=5.0, installed: 8.1.3]
      - colorama [required: Any, installed: 0.4.6]
    - cloudpickle [required: >=0.4, installed: 2.2.0]
    - dask [required: Any, installed: 2022.12.0]
      - click [required: >=7.0, installed: 8.1.3]
        - colorama [required: Any, installed: 0.4.6]
      - cloudpickle [required: >=1.1.1, installed: 2.2.0]
      - fsspec [required: >=0.6.0, installed: 2022.11.0]
      - packaging [required: >=20.0, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - partd [required: >=0.3.10, installed: 1.3.0]
        - locket [required: Any, installed: 1.0.0]
        - toolz [required: Any, installed: 0.12.0]
      - pyyaml [required: >=5.3.1, installed: 6.0]
      - toolz [required: >=0.8.2, installed: 0.12.0]
    - distributed [required: Any, installed: 2022.12.0]
      - click [required: >=7.0, installed: 8.1.3]
        - colorama [required: Any, installed: 0.4.6]
      - cloudpickle [required: >=1.5.0, installed: 2.2.0]
      - dask [required: ==2022.12.0, installed: 2022.12.0]
        - click [required: >=7.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
        - cloudpickle [required: >=1.1.1, installed: 2.2.0]
        - fsspec [required: >=0.6.0, installed: 2022.11.0]
        - packaging [required: >=20.0, installed: 21.3]
          - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
        - partd [required: >=0.3.10, installed: 1.3.0]
          - locket [required: Any, installed: 1.0.0]
          - toolz [required: Any, installed: 0.12.0]
        - pyyaml [required: >=5.3.1, installed: 6.0]
        - toolz [required: >=0.8.2, installed: 0.12.0]
      - jinja2 [required: Any, installed: 3.1.2]
        - MarkupSafe [required: >=2.0, installed: 2.1.1]
      - locket [required: >=1.0.0, installed: 1.0.0]
      - msgpack [required: >=0.6.0, installed: 1.0.4]
      - packaging [required: >=20.0, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - psutil [required: >=5.0, installed: 5.9.4]
      - pyyaml [required: Any, installed: 6.0]
      - sortedcontainers [required: !=2.0.1,!=2.0.0, installed: 2.4.0]
      - tblib [required: >=1.6.0, installed: 1.7.0]
      - toolz [required: >=0.10.0, installed: 0.12.0]
      - tornado [required: >=6.0.3, installed: 6.2]
      - urllib3 [required: Any, installed: 1.26.13]
      - zict [required: >=0.1.3, installed: 2.2.0]
        - heapdict [required: Any, installed: 1.0.1]
    - GeoAlchemy2 [required: Any, installed: 0.12.5]
      - packaging [required: Any, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - SQLAlchemy [required: >=1.4, installed: 1.4.44]
        - greenlet [required: !=0.4.17, installed: 2.0.1]
    - jsonschema [required: Any, installed: 4.17.3]
      - attrs [required: >=17.4.0, installed: 22.1.0]
      - pyrsistent [required: >=0.14.0,!=0.17.2,!=0.17.1,!=0.17.0, installed: 0.19.2]
    - lark [required: Any, installed: 1.1.5]
    - netcdf4 [required: Any, installed: 1.6.2]
      - cftime [required: Any, installed: 1.6.2]
        - numpy [required: >1.13.3, installed: 1.24.1]
      - numpy [required: >=1.9, installed: 1.24.1]
    - numpy [required: Any, installed: 1.24.1]
    - pandas [required: Any, installed: 1.5.2]
      - numpy [required: >=1.20.3, installed: 1.24.1]
      - python-dateutil [required: >=2.8.1, installed: 2.8.2]
        - six [required: >=1.5, installed: 1.16.0]
      - pytz [required: >=2020.1, installed: 2022.6]
    - psycopg2 [required: Any, installed: 2.9.5]
    - pyproj [required: >=2.5, installed: 3.4.0]
      - certifi [required: Any, installed: 2022.12.7]
    - python-dateutil [required: Any, installed: 2.8.2]
      - six [required: >=1.5, installed: 1.16.0]
    - pyyaml [required: Any, installed: 6.0]
    - rasterio [required: >=1.3.2, installed: 1.3.4]
      - affine [required: Any, installed: 2.3.1]
      - attrs [required: Any, installed: 22.1.0]
      - certifi [required: Any, installed: 2022.12.7]
      - click [required: >=4.0, installed: 8.1.3]
        - colorama [required: Any, installed: 0.4.6]
      - click-plugins [required: Any, installed: 1.1.1]
        - click [required: >=4.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
      - cligj [required: >=0.5, installed: 0.7.2]
        - click [required: >=4.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
      - numpy [required: >=1.18, installed: 1.24.1]
      - setuptools [required: Any, installed: 65.5.1]
      - snuggs [required: >=1.4.1, installed: 1.4.7]
        - numpy [required: Any, installed: 1.24.1]
        - pyparsing [required: >=2.1.6, installed: 3.0.9]
    - shapely [required: >=1.6.4, installed: 1.8.4]
    - sqlalchemy [required: Any, installed: 1.4.44]
      - greenlet [required: !=0.4.17, installed: 2.0.1]
    - toolz [required: Any, installed: 0.12.0]
    - xarray [required: >=0.9,!=2022.6.0, installed: 2022.12.0]
      - numpy [required: >=1.20, installed: 1.24.1]
      - packaging [required: >=21.3, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - pandas [required: >=1.3, installed: 1.5.2]
        - numpy [required: >=1.20.3, installed: 1.24.1]
        - python-dateutil [required: >=2.8.1, installed: 2.8.2]
          - six [required: >=1.5, installed: 1.16.0]
        - pytz [required: >=2020.1, installed: 2022.6]
  - geopandas [required: >=0.11.1,<1, installed: 0.12.1]
    - fiona [required: >=1.8, installed: 1.8.22]
      - attrs [required: >=17, installed: 22.1.0]
      - certifi [required: Any, installed: 2022.12.7]
      - click [required: >=4.0, installed: 8.1.3]
        - colorama [required: Any, installed: 0.4.6]
      - click-plugins [required: >=1.0, installed: 1.1.1]
        - click [required: >=4.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
      - cligj [required: >=0.5, installed: 0.7.2]
        - click [required: >=4.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
      - munch [required: Any, installed: 2.5.0]
        - six [required: Any, installed: 1.16.0]
      - setuptools [required: Any, installed: 65.5.1]
      - six [required: >=1.7, installed: 1.16.0]
    - packaging [required: Any, installed: 21.3]
      - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
    - pandas [required: >=1.0.0, installed: 1.5.2]
      - numpy [required: >=1.20.3, installed: 1.24.1]
      - python-dateutil [required: >=2.8.1, installed: 2.8.2]
        - six [required: >=1.5, installed: 1.16.0]
      - pytz [required: >=2020.1, installed: 2022.6]
    - pyproj [required: >=2.6.1.post1, installed: 3.4.0]
      - certifi [required: Any, installed: 2022.12.7]
    - shapely [required: >=1.7, installed: 1.8.4]
  - odc-algo [required: >=0.2.3,<1, installed: 0.2.3]
    - affine [required: Any, installed: 2.3.1]
    - dask [required: Any, installed: 2022.12.0]
      - click [required: >=7.0, installed: 8.1.3]
        - colorama [required: Any, installed: 0.4.6]
      - cloudpickle [required: >=1.1.1, installed: 2.2.0]
      - fsspec [required: >=0.6.0, installed: 2022.11.0]
      - packaging [required: >=20.0, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - partd [required: >=0.3.10, installed: 1.3.0]
        - locket [required: Any, installed: 1.0.0]
        - toolz [required: Any, installed: 0.12.0]
      - pyyaml [required: >=5.3.1, installed: 6.0]
      - toolz [required: >=0.8.2, installed: 0.12.0]
    - dask-image [required: Any, installed: 2022.9.0]
      - dask [required: >=0.16.1, installed: 2022.12.0]
        - click [required: >=7.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
        - cloudpickle [required: >=1.1.1, installed: 2.2.0]
        - fsspec [required: >=0.6.0, installed: 2022.11.0]
        - packaging [required: >=20.0, installed: 21.3]
          - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
        - partd [required: >=0.3.10, installed: 1.3.0]
          - locket [required: Any, installed: 1.0.0]
          - toolz [required: Any, installed: 0.12.0]
        - pyyaml [required: >=5.3.1, installed: 6.0]
        - toolz [required: >=0.8.2, installed: 0.12.0]
      - numpy [required: >=1.11.3, installed: 1.24.1]
      - pims [required: >=0.4.1, installed: 0.6.1]
        - imageio [required: Any, installed: 2.22.4]
          - numpy [required: Any, installed: 1.24.1]
          - pillow [required: >=8.3.2, installed: 9.2.0]
        - numpy [required: >=1.19, installed: 1.24.1]
        - slicerator [required: >=0.9.8, installed: 1.1.0]
      - scipy [required: >=0.19.1, installed: 1.9.3]
        - numpy [required: >=1.18.5,<1.26.0, installed: 1.24.1]
    - datacube [required: >=1.8.5, installed: 1.8.9]
      - affine [required: Any, installed: 2.3.1]
      - cachetools [required: Any, installed: 5.2.0]
      - click [required: >=5.0, installed: 8.1.3]
        - colorama [required: Any, installed: 0.4.6]
      - cloudpickle [required: >=0.4, installed: 2.2.0]
      - dask [required: Any, installed: 2022.12.0]
        - click [required: >=7.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
        - cloudpickle [required: >=1.1.1, installed: 2.2.0]
        - fsspec [required: >=0.6.0, installed: 2022.11.0]
        - packaging [required: >=20.0, installed: 21.3]
          - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
        - partd [required: >=0.3.10, installed: 1.3.0]
          - locket [required: Any, installed: 1.0.0]
          - toolz [required: Any, installed: 0.12.0]
        - pyyaml [required: >=5.3.1, installed: 6.0]
        - toolz [required: >=0.8.2, installed: 0.12.0]
      - distributed [required: Any, installed: 2022.12.0]
        - click [required: >=7.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
        - cloudpickle [required: >=1.5.0, installed: 2.2.0]
        - dask [required: ==2022.12.0, installed: 2022.12.0]
          - click [required: >=7.0, installed: 8.1.3]
            - colorama [required: Any, installed: 0.4.6]
          - cloudpickle [required: >=1.1.1, installed: 2.2.0]
          - fsspec [required: >=0.6.0, installed: 2022.11.0]
          - packaging [required: >=20.0, installed: 21.3]
            - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
          - partd [required: >=0.3.10, installed: 1.3.0]
            - locket [required: Any, installed: 1.0.0]
            - toolz [required: Any, installed: 0.12.0]
          - pyyaml [required: >=5.3.1, installed: 6.0]
          - toolz [required: >=0.8.2, installed: 0.12.0]
        - jinja2 [required: Any, installed: 3.1.2]
          - MarkupSafe [required: >=2.0, installed: 2.1.1]
        - locket [required: >=1.0.0, installed: 1.0.0]
        - msgpack [required: >=0.6.0, installed: 1.0.4]
        - packaging [required: >=20.0, installed: 21.3]
          - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
        - psutil [required: >=5.0, installed: 5.9.4]
        - pyyaml [required: Any, installed: 6.0]
        - sortedcontainers [required: !=2.0.1,!=2.0.0, installed: 2.4.0]
        - tblib [required: >=1.6.0, installed: 1.7.0]
        - toolz [required: >=0.10.0, installed: 0.12.0]
        - tornado [required: >=6.0.3, installed: 6.2]
        - urllib3 [required: Any, installed: 1.26.13]
        - zict [required: >=0.1.3, installed: 2.2.0]
          - heapdict [required: Any, installed: 1.0.1]
      - GeoAlchemy2 [required: Any, installed: 0.12.5]
        - packaging [required: Any, installed: 21.3]
          - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
        - SQLAlchemy [required: >=1.4, installed: 1.4.44]
          - greenlet [required: !=0.4.17, installed: 2.0.1]
      - jsonschema [required: Any, installed: 4.17.3]
        - attrs [required: >=17.4.0, installed: 22.1.0]
        - pyrsistent [required: >=0.14.0,!=0.17.2,!=0.17.1,!=0.17.0, installed: 0.19.2]
      - lark [required: Any, installed: 1.1.5]
      - netcdf4 [required: Any, installed: 1.6.2]
        - cftime [required: Any, installed: 1.6.2]
          - numpy [required: >1.13.3, installed: 1.24.1]
        - numpy [required: >=1.9, installed: 1.24.1]
      - numpy [required: Any, installed: 1.24.1]
      - pandas [required: Any, installed: 1.5.2]
        - numpy [required: >=1.20.3, installed: 1.24.1]
        - python-dateutil [required: >=2.8.1, installed: 2.8.2]
          - six [required: >=1.5, installed: 1.16.0]
        - pytz [required: >=2020.1, installed: 2022.6]
      - psycopg2 [required: Any, installed: 2.9.5]
      - pyproj [required: >=2.5, installed: 3.4.0]
        - certifi [required: Any, installed: 2022.12.7]
      - python-dateutil [required: Any, installed: 2.8.2]
        - six [required: >=1.5, installed: 1.16.0]
      - pyyaml [required: Any, installed: 6.0]
      - rasterio [required: >=1.3.2, installed: 1.3.4]
        - affine [required: Any, installed: 2.3.1]
        - attrs [required: Any, installed: 22.1.0]
        - certifi [required: Any, installed: 2022.12.7]
        - click [required: >=4.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
        - click-plugins [required: Any, installed: 1.1.1]
          - click [required: >=4.0, installed: 8.1.3]
            - colorama [required: Any, installed: 0.4.6]
        - cligj [required: >=0.5, installed: 0.7.2]
          - click [required: >=4.0, installed: 8.1.3]
            - colorama [required: Any, installed: 0.4.6]
        - numpy [required: >=1.18, installed: 1.24.1]
        - setuptools [required: Any, installed: 65.5.1]
        - snuggs [required: >=1.4.1, installed: 1.4.7]
          - numpy [required: Any, installed: 1.24.1]
          - pyparsing [required: >=2.1.6, installed: 3.0.9]
      - shapely [required: >=1.6.4, installed: 1.8.4]
      - sqlalchemy [required: Any, installed: 1.4.44]
        - greenlet [required: !=0.4.17, installed: 2.0.1]
      - toolz [required: Any, installed: 0.12.0]
      - xarray [required: >=0.9,!=2022.6.0, installed: 2022.12.0]
        - numpy [required: >=1.20, installed: 1.24.1]
        - packaging [required: >=21.3, installed: 21.3]
          - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
        - pandas [required: >=1.3, installed: 1.5.2]
          - numpy [required: >=1.20.3, installed: 1.24.1]
          - python-dateutil [required: >=2.8.1, installed: 2.8.2]
            - six [required: >=1.5, installed: 1.16.0]
          - pytz [required: >=2020.1, installed: 2022.6]
    - distributed [required: Any, installed: 2022.12.0]
      - click [required: >=7.0, installed: 8.1.3]
        - colorama [required: Any, installed: 0.4.6]
      - cloudpickle [required: >=1.5.0, installed: 2.2.0]
      - dask [required: ==2022.12.0, installed: 2022.12.0]
        - click [required: >=7.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
        - cloudpickle [required: >=1.1.1, installed: 2.2.0]
        - fsspec [required: >=0.6.0, installed: 2022.11.0]
        - packaging [required: >=20.0, installed: 21.3]
          - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
        - partd [required: >=0.3.10, installed: 1.3.0]
          - locket [required: Any, installed: 1.0.0]
          - toolz [required: Any, installed: 0.12.0]
        - pyyaml [required: >=5.3.1, installed: 6.0]
        - toolz [required: >=0.8.2, installed: 0.12.0]
      - jinja2 [required: Any, installed: 3.1.2]
        - MarkupSafe [required: >=2.0, installed: 2.1.1]
      - locket [required: >=1.0.0, installed: 1.0.0]
      - msgpack [required: >=0.6.0, installed: 1.0.4]
      - packaging [required: >=20.0, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - psutil [required: >=5.0, installed: 5.9.4]
      - pyyaml [required: Any, installed: 6.0]
      - sortedcontainers [required: !=2.0.1,!=2.0.0, installed: 2.4.0]
      - tblib [required: >=1.6.0, installed: 1.7.0]
      - toolz [required: >=0.10.0, installed: 0.12.0]
      - tornado [required: >=6.0.3, installed: 6.2]
      - urllib3 [required: Any, installed: 1.26.13]
      - zict [required: >=0.1.3, installed: 2.2.0]
        - heapdict [required: Any, installed: 1.0.1]
    - numexpr [required: Any, installed: 2.8.4]
      - numpy [required: >=1.13.3, installed: 1.24.1]
    - numpy [required: Any, installed: 1.24.1]
    - rasterio [required: >=1.3.2, installed: 1.3.4]
      - affine [required: Any, installed: 2.3.1]
      - attrs [required: Any, installed: 22.1.0]
      - certifi [required: Any, installed: 2022.12.7]
      - click [required: >=4.0, installed: 8.1.3]
        - colorama [required: Any, installed: 0.4.6]
      - click-plugins [required: Any, installed: 1.1.1]
        - click [required: >=4.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
      - cligj [required: >=0.5, installed: 0.7.2]
        - click [required: >=4.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
      - numpy [required: >=1.18, installed: 1.24.1]
      - setuptools [required: Any, installed: 65.5.1]
      - snuggs [required: >=1.4.1, installed: 1.4.7]
        - numpy [required: Any, installed: 1.24.1]
        - pyparsing [required: >=2.1.6, installed: 3.0.9]
    - scikit-image [required: Any, installed: 0.19.3]
      - imageio [required: >=2.4.1, installed: 2.22.4]
        - numpy [required: Any, installed: 1.24.1]
        - pillow [required: >=8.3.2, installed: 9.2.0]
      - networkx [required: >=2.2, installed: 2.8.8]
      - numpy [required: >=1.17.0, installed: 1.24.1]
      - packaging [required: >=20.0, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - pillow [required: >=6.1.0,!=8.3.0,!=7.1.1,!=7.1.0, installed: 9.2.0]
      - PyWavelets [required: >=1.1.1, installed: 1.4.1]
        - numpy [required: >=1.17.3, installed: 1.24.1]
      - scipy [required: >=1.4.1, installed: 1.9.3]
        - numpy [required: >=1.18.5,<1.26.0, installed: 1.24.1]
      - tifffile [required: >=2019.7.26, installed: 2022.10.10]
        - numpy [required: >=1.19.2, installed: 1.24.1]
    - toolz [required: Any, installed: 0.12.0]
    - xarray [required: Any, installed: 2022.12.0]
      - numpy [required: >=1.20, installed: 1.24.1]
      - packaging [required: >=21.3, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - pandas [required: >=1.3, installed: 1.5.2]
        - numpy [required: >=1.20.3, installed: 1.24.1]
        - python-dateutil [required: >=2.8.1, installed: 2.8.2]
          - six [required: >=1.5, installed: 1.16.0]
        - pytz [required: >=2020.1, installed: 2022.6]
  - odc-geo [required: >=0.3.2,<0.4.0, installed: 0.3.3]
    - affine [required: Any, installed: 2.3.1]
    - cachetools [required: Any, installed: 5.2.0]
    - numpy [required: Any, installed: 1.24.1]
    - pyproj [required: >=3.0.0, installed: 3.4.0]
      - certifi [required: Any, installed: 2022.12.7]
    - shapely [required: Any, installed: 1.8.4]
  - openeo-pg-parser-networkx [required: >=2023.1.2, installed: 2023.1.2]
    - geojson-pydantic [required: >=0.5.0,<0.6.0, installed: 0.5.0]
      - pydantic [required: Any, installed: 1.10.2]
        - typing-extensions [required: >=4.1.0, installed: 4.4.0]
    - networkx [required: >=2.8.6,<3.0.0, installed: 2.8.8]
    - numpy [required: >=1.24.1,<2.0.0, installed: 1.24.1]
    - pendulum [required: >=2.1.2,<3.0.0, installed: 2.1.2]
      - python-dateutil [required: >=2.6,<3.0, installed: 2.8.2]
        - six [required: >=1.5, installed: 1.16.0]
      - pytzdata [required: >=2020.1, installed: 2020.1]
    - pydantic [required: >=1.9.1,<2.0.0, installed: 1.10.2]
      - typing-extensions [required: >=4.1.0, installed: 4.4.0]
    - pyproj [required: >=3.4.0,<4.0.0, installed: 3.4.0]
      - certifi [required: Any, installed: 2022.12.7]
    - shapely [required: >=1.8, installed: 1.8.4]
  - rasterio [required: >=1.3.4,<2.0.0, installed: 1.3.4]
    - affine [required: Any, installed: 2.3.1]
    - attrs [required: Any, installed: 22.1.0]
    - certifi [required: Any, installed: 2022.12.7]
    - click [required: >=4.0, installed: 8.1.3]
      - colorama [required: Any, installed: 0.4.6]
    - click-plugins [required: Any, installed: 1.1.1]
      - click [required: >=4.0, installed: 8.1.3]
        - colorama [required: Any, installed: 0.4.6]
    - cligj [required: >=0.5, installed: 0.7.2]
      - click [required: >=4.0, installed: 8.1.3]
        - colorama [required: Any, installed: 0.4.6]
    - numpy [required: >=1.18, installed: 1.24.1]
    - setuptools [required: Any, installed: 65.5.1]
    - snuggs [required: >=1.4.1, installed: 1.4.7]
      - numpy [required: Any, installed: 1.24.1]
      - pyparsing [required: >=2.1.6, installed: 3.0.9]
  - rioxarray [required: >=0.12.0,<1, installed: 0.13.1]
    - packaging [required: Any, installed: 21.3]
      - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
    - pyproj [required: >=2.2, installed: 3.4.0]
      - certifi [required: Any, installed: 2022.12.7]
    - rasterio [required: >=1.1.1, installed: 1.3.4]
      - affine [required: Any, installed: 2.3.1]
      - attrs [required: Any, installed: 22.1.0]
      - certifi [required: Any, installed: 2022.12.7]
      - click [required: >=4.0, installed: 8.1.3]
        - colorama [required: Any, installed: 0.4.6]
      - click-plugins [required: Any, installed: 1.1.1]
        - click [required: >=4.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
      - cligj [required: >=0.5, installed: 0.7.2]
        - click [required: >=4.0, installed: 8.1.3]
          - colorama [required: Any, installed: 0.4.6]
      - numpy [required: >=1.18, installed: 1.24.1]
      - setuptools [required: Any, installed: 65.5.1]
      - snuggs [required: >=1.4.1, installed: 1.4.7]
        - numpy [required: Any, installed: 1.24.1]
        - pyparsing [required: >=2.1.6, installed: 3.0.9]
    - xarray [required: >=0.17, installed: 2022.12.0]
      - numpy [required: >=1.20, installed: 1.24.1]
      - packaging [required: >=21.3, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - pandas [required: >=1.3, installed: 1.5.2]
        - numpy [required: >=1.20.3, installed: 1.24.1]
        - python-dateutil [required: >=2.8.1, installed: 2.8.2]
          - six [required: >=1.5, installed: 1.16.0]
        - pytz [required: >=2020.1, installed: 2022.6]
  - xarray [required: >=2022.11.0, installed: 2022.12.0]
    - numpy [required: >=1.20, installed: 1.24.1]
    - packaging [required: >=21.3, installed: 21.3]
      - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
    - pandas [required: >=1.3, installed: 1.5.2]
      - numpy [required: >=1.20.3, installed: 1.24.1]
      - python-dateutil [required: >=2.8.1, installed: 2.8.2]
        - six [required: >=1.5, installed: 1.16.0]
      - pytz [required: >=2020.1, installed: 2022.6]
  - xgboost [required: >=1.5.1,<2.0.0, installed: 1.7.1]
    - numpy [required: Any, installed: 1.24.1]
    - scipy [required: Any, installed: 1.9.3]
      - numpy [required: >=1.18.5,<1.26.0, installed: 1.24.1]
```

**Comment by LukeWeidenwalker** [comment]
Hey - sorry for the late reply on this, I was on sick leave last week!

IMHO odc-algo is the worst offender here. It's not a stable/well-defined library in the first place and it pulls in 1) datacube + all its sql stuff we don't use here and 2) scikit-image, which we don't use at all.
We literally only need it for its xr_reproject method, so finding an alternative to this would probably be the biggest "bang for your buck" in terms of effort vs. reduced dependency set.

There's already an open issue for porting that functionality into odc-geo, could be a good contribution over there: https://github.com/opendatacube/odc-geo/issues/26

**Comment by LukeWeidenwalker** [comment]
As discussed today, we'll put `resample_spatial` into the experimental folder and expose a new build variant `[experimental]` that we'll use internally. A vanilla installation of this repo should then not include these unnecessary dependencies from odc-algo!

**Comment by clausmichele** [comment]
Thanks Lukas. Little note: it's `resample_cube_spatial`, `resample_spatial` is currently not yet implemented here.

**Comment by LukeWeidenwalker** [comment]
@clausmichele just released 2023.3.0, this has the discussed build variants and doesn't pull in matplotlib from the parser anymore! 👍
